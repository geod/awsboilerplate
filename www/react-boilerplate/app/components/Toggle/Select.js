import styled from 'styled-components';

const Select = styled.select`
  line-height: 1em;
  background-color: transparent;
  border-style: none;
`;

export default Select;
