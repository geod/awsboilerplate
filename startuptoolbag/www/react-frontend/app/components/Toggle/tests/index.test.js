import React from 'react';
import { render } from 'react-testing-library';
import { IntlProvider, defineMessages } from 'react-intl';

import Toggle from '../index';

describe('<Toggle />', () => {
  it('should contain default text', () => {
    const defaultEnMessage = 'someContent';
    const defaultDeMessage = 'someOtherContent';
    const messages = defineMessages({
      en: {
        id: 'boilerplate.containers.LocaleToggle.en',
        defaultMessage: defaultEnMessage,
      },
      de: {
        id: 'boilerplate.containers.LocaleToggle.en',
        defaultMessage: defaultDeMessage,
      },
    });
    const { container } = render(
      <IntlProvider locale="en">
        <Toggle values={['en', 'de']} messages={messages} />
      </IntlProvider>,
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should not have ToggleOptions if props.values is not defined', () => {
    const { container } = render(<Toggle />);
    const elements = container.querySelectorAll('option');
    expect(elements).toHaveLength(1);
    expect(container.firstChild.value).toEqual('--');
  });
});
