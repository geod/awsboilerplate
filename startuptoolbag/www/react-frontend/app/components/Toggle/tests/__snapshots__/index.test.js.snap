// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Toggle /> should contain default text 1`] = `
<select
  class="Select-sc-1sm01tk-0 wIjFa"
>
  <option
    value="en"
  >
    someContent
  </option>
  <option
    value="de"
  >
    someOtherContent
  </option>
</select>
`;
